api/v1alpha1/argocdcommitstatus_types.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// ArgoCDCommitStatusSpec defines the desired state of ArgoCDCommitStatus.
+type ArgoCDCommitStatusSpec struct {
+	// +kubebuilder:validation:Required
+	PromotionStrategyRef ObjectReference `json:"promotionStrategyRef,omitempty"`
+
+	// +kubebuilder:validation:Required
+	ApplicationSelector *metav1.LabelSelector `json:"applicationSelector,omitempty"`
+}
+
+// ArgoCDCommitStatusStatus defines the observed state of ArgoCDCommitStatus.
+type ArgoCDCommitStatusStatus struct {
+	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+	ApplicationsSelected []SelectedApplications `json:"applicationsSelected,omitempty"`
+}
+
+type SelectedApplications struct {
+	Namespace string            `json:"namespace"`
+	Name      string            `json:"name"`
+	Sate      CommitStatusPhase `json:"state"`
```

Comment:
```suggestion
	State      CommitStatusPhase `json:"state"`
```

Should we actually just rename it phase?